documentation/docs/discovery.mdx

```diff
@@ -1,127 +1,106 @@
-`fragment` trimmed away.
+However, in the second example Alice is the payee but her wallet still uses
+Bob's Payment Pointer to discover his account and create and authorize a mandate
+to pull funds from it.
 
-E.g. Given the Payment Pointer `$wallet.example/alice` the subject identifier is
-`https://wallet.example/alice` and the issuer identifier is
-`https://wallet.example`.
+<RequestToPay />
 
-Once the client has determined the issuer identifier it can retrieve the server
-metadata for the issuer.
+Discovery starts with an identifier, which in Open Payments is an account
+identifier called a **Payment Pointer**.
 
-## Metadata
-
-The metadata is a JSON document resource that describes the various service
-endpoints that are available to:
-
-- connect to the `subject`'s account via Interledger
-- get authorization to perform actions on the `subject`'s account (using OAuth
-  2.0)
-- get more information about the `subject` (using OpenID Connect)
-- create agreements with the `subject`
-
-The document is hosted at a `.well-known` URL relative to the issuer identifier.
+## Payment Pointers
 
-[RFC8414](https://tools.ietf.org/html/rfc8414) defines the default well-known
-URI string to be `/.well-known/oauth-authorization-server`.
+All entities in Open Payments are **resources** identified by, and addressable
+via, URLs in keeping with the
+[REST](https://tools.ietf.org/html/rfc7231#section-2) pattern. This includes
+[accounts](./accounts) which are the root resource in Open Payments. Discovery
+involves finding the URL of the account resource because all other resource URLs
+are relative to the account.
 
-It also states the following:
+> Example account URL:  
+> `https://wallet.example/accounts/dee69df8-be40-478f-9828-6267f4d42b01`
 
-> Different applications utilizing OAuth authorization servers in
-> application-specific ways may define and register different well- known URI
-> suffixes used to publish authorization server metadata as used by those
-> applications.
+To make account URLs shorter and easier to transcribe they can be encoded as a
+**[Payment Pointer](https://paymentpointers.org)**. In this form they also have
+the advantage of being easily and unambiguously recognizable as Payment
+Pointers.
 
-Open Payments defines the well-known URL `/.well-known/open-payments`. The meta-data document MUST be queried using an HTTP "GET" request at the
-previously specified path.
+> Example Payment Pointer:  
+> `$wallet.example/accounts/dee69df8-be40-478f-9828-6267f4d42b01`
 
-The client would make the following request when the issuer identifier is
-`https://wallet.example` and the well-known URI suffix is `open-payments` to
-obtain the metadata, since the issuer identifier contains no path component:
+[Wallets](./terminology#wallet) that support Open Payments will define one or
+more unique account URLs for each account they service on behalf of the account
+owner.
 
-```http
-GET /.well-known/open-payments HTTP/1.1
```

Comment:
Overall, I like the idea of providing account-level meta-data the way this PR is moving. However, I have a few questions around how this might work with alternate subjects types. While this isn't likely a concern if the `subject` of an invoice is always a PaymentPointer, if we ever wanted to consider an alternate subject, we might have problems bootstrapping the OpenPayments endpoints.

Is the idea just that were such an alternate identifier to be used, we would want to define OpenPayments endpoints in that protocol as opposed to defining them in the open-payment server's host-wide meta data?

I'm trying to think how, given an identifier like (e.g.) `david@gmail.com` how I would discover the OpenPayments endpoints. Per my comment above, is the thinking just that someone would need to define a discovery protocol for email addresses (e.g., [Webfinger](https://webfinger.net)) and one of the things advertised in that meta-data mechanism would be OpenPayments endpoints?

Reply:
Open Payments APIs are rooted in an account URL.
Open Payments defines a single discovery protocol which takes a Payment Pointer as input and results in an account URL.
If you want to start with a different identifier then you'd need to define a discovery protocol that uses that to also end with an account URL.